Content.Server/GameObjects/Components/Atmos/GasAnalyzerComponent.cs

```diff
@@ -1,40 +1,59 @@
 ﻿#nullable enable
 using Content.Server.GameObjects.EntitySystems;
+using Content.Server.Interfaces;
 using Content.Shared.Atmos;
 using Content.Shared.GameObjects.EntitySystems;
+using Content.Shared.Interfaces;
+using Content.Shared.Interfaces.GameObjects.Components;
+using Content.Shared.Utility;
 using Robust.Shared.GameObjects;
 using Robust.Shared.GameObjects.Systems;
+using Robust.Shared.IoC;
 using Robust.Shared.Map;
 using Robust.Shared.Utility;
 
 namespace Content.Server.GameObjects.Components.Atmos
 {
     [RegisterComponent]
-    public class GasAnalyzerComponent : Component, IExamine
+    public class GasAnalyzerComponent : Component, IAfterInteract
     {
+#pragma warning disable 649
+        [Dependency] private IServerNotifyManager _notifyManager = default!;
+#pragma warning restore 649
         public override string Name => "GasAnalyzer";
-        public void Examine(FormattedMessage message, bool inDetailsRange)
+
+        void IAfterInteract.AfterInteract(AfterInteractEventArgs eventArgs)
         {
-            if (!inDetailsRange) return;
+            if (!eventArgs.CanReach)
+            {
+                _notifyManager.PopupMessage(eventArgs.ClickLocation, eventArgs.User, "Can't reach");
+                return;
+            }
 
             var gam = EntitySystem.Get<AtmosphereSystem>().GetGridAtmosphere(Owner.Transform.GridID);
 
-            var tile = gam?.GetTile(Owner.Transform.GridPosition).Air;
+            var tile = gam?.GetTile(eventArgs.ClickLocation).Air;
 
-            if (tile == null) return;
+            if (tile == null)
+            {
+                _notifyManager.PopupMessage(eventArgs.ClickLocation, eventArgs.User, "No atmosphere!");
+                return;
+            }
 
-            message.AddText($"Pressure: {tile.Pressure}\n");
-            message.AddText($"Temperature: {tile.Temperature}\n");
+            string message = "";
+            message += $"Pressure: {tile.Pressure:0.##} kPa\n";
+            message += $"Temperature: {tile.Temperature}K ({TemperatureHelpers.KelvinToCelsius(tile.Temperature)}°C)\n";
 
             for (int i = 0; i < Atmospherics.TotalNumberOfGases; i++)
             {
                 var gas = Atmospherics.GetGas(i);
 
                 if (tile.Gases[i] <= Atmospherics.GasMinMoles) continue;
 
-                message.AddText(gas.Name);
-                message.AddText($"\n Moles: {tile.Gases[i]}\n");
+                message += $"{gas.Name}: {tile.Gases[i]} mol \n";
             }
+            message = message.TrimEnd(); // Remove annoying newline
```

Comment:
If you want, you can get rid of this by not adding `\n` for the temperature line, and add the `\n` at the beginning of each gas line.